serverless/internal/storage/fs/fs.go

```diff
@@ -246,12 +247,76 @@ func (fs *Storage) ScanSequenced(begin uint64, f func(seq uint64, entry []byte)
 // If no complete tile exists at that location, it will attempt to find a
 // partial tile for the given tree size at that location.
 func (fs *Storage) GetTile(level, index, logSize uint64) (*api.Tile, error) {
-	return nil, errors.New("unimplemented")
+	tileSize := storage.PartialTileSize(level, index, logSize)
+	p := filepath.Join(layout.TilePath(fs.rootDir, level, index, tileSize))
+	t, err := ioutil.ReadFile(p)
+	if err != nil && !errors.Is(err, os.ErrNotExist) {
+		return nil, fmt.Errorf("failed to read tile at %q: %w", p, err)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	var tile api.Tile
+	if err := json.Unmarshal(t, &tile); err != nil {
+		return nil, fmt.Errorf("failed to parse tile: %w", err)
+	}
+	return &tile, nil
 }
 
 // StoreTile writes a tile out to disk.
+// Fully populated tiles are stored at the path corresponding to the level &
+// index parameters, partially populated (i.e. right-hand edge) tiles are
+// stored with a .xx suffix where xx is the number of "tile leaves" in hex.
 func (fs *Storage) StoreTile(level, index uint64, tile *api.Tile) error {
-	return errors.New("unimplemented")
+	tileSize := uint64(tile.NumLeaves)
+	glog.V(2).Infof("StoreTile: level %d index %x ts: %x", level, index, tileSize)
+	if tileSize == 0 || tileSize > 256 {
+		return fmt.Errorf("tileSize %d must be > 0 and <= 256", tileSize)
+	}
+	t, err := json.Marshal(tile)
+	if err != nil {
+		return fmt.Errorf("failed to marshal tile: %w", err)
+	}
+
+	tDir, tFile := layout.TilePath(fs.rootDir, level, index, uint64(tileSize%256))
```

Comment:
Why is the mod 256 needed here? We just asserted it was in range [0, 256).

Reply:
Well, we asserted that it's [1,256] - i.e. not-empty, and not bigger than 2^8.
The logic in `TilePath` is that full tiles are named `ID` and partial tiles named `ID.xx`, so it takes a param with the number of leaves expected for partial tiles and uses that for .xx iff it's > 0.

I've updated the `TilePath` param name and func comment to help make that clearer.